frontend/public/components/graphs/resizable.tsx

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable no-undef, no-unused-vars */
+import * as React from 'react';
+
+export const resizable = (Component) => {
```

Comment:
I'm thinking `resizable` could be replaced by a custom hook. Something like

```
export const useRefWidth = () => {
  const ref = React.useRef(null);
  const [width, setWidth] = React.useState(0);
  const handleResize = () => setWidth(ref.current.clientWidth);

  React.useEffect(() => {
    window.addEventListener('resize', handleResize);
    window.addEventListener('nav_toggle', handleResize);
    handleResize();

    return () => {
      window.removeEventListener('resize', handleResize);
      window.removeEventListener('nav_toggle', handleResize);
    };
  }, []);

  return [ref, width];
};
```

Then call `[containerRef, width] = useRefWidth()` in the `Line` component, so we wouldn't need this HoC at all.

Reply:
Yeah, I'm definitely good with that. Had a lot of other design issues to address, but I can do this as well.